### PR TITLE
magento/magento2#26575: REST API rest/all/V1/orders/create does not trigger "inventory_reservations_placement" plugin

### LINKED ISSUE #3185: REST API rest/all/V1/orders/create does not trigger "inventory_reservations_placement" plugin
REST API `rest/all/V1/orders/create` does not trigger `inventory_reservations_placement` plugin.

### Preconditions (*)

`Magento 2.3.3` with `sample` data

1. Create OAuth tokens with access to create order resource

2. End point: `https://INSTANCE/rest/all/V1/orders/create`
3. Method: `PUT`
4. Content-Type: application/json
5. JSON data to verify:

```
{
  "entity": {
    "base_currency_code": "USD",
    "base_discount_amount": 0,
    "base_discount_tax_compensation_amount": 0,
    "base_grand_total": 53.71,
    "base_shipping_amount": 5,
    "base_shipping_discount_amount": null,
    "base_shipping_incl_tax": null,
    "base_shipping_tax_amount": null,
    "base_subtotal": 45,
    "base_subtotal_incl_tax": 48.71,
    "base_tax_amount": 3.71,
    "base_to_global_rate": 1,
    "base_to_order_rate": 1,
    "base_total_due": 53.71,
    "base_total_paid": 0,
    "billing_address": {
      "address_type": "billing",
      "city": "Calder",
      "company": "",
      "country_id": "US",
      "customer_address_id": null,
      "email": "roni_cost@example.com",
      "entity_id": null,
      "firstname": "Veronica",
      "lastname": "Costello",
      "parent_id": null,
      "postcode": "49628-7978",
      "region": "MI",
      "region_code": null,
      "street": [
        "6146 Honey Bluff Parkway", ""
      ],
      "telephone": "0000000000"
    },
    "billing_address_id": null,
    "created_at": null,
    "customer_email": "roni_cost@example.com",
    "customer_firstname": "Veronica",
    "customer_group_id": 1,
    "customer_id": 1,
    "customer_is_guest": 0,
    "customer_lastname": "Costello",
    "customer_note_notify": null,
    "discount_amount": 0,
    "discount_tax_compensation_amount": 0,
    "email_sent": null,
    "entity_id": null,
    "extension_attributes": {
      "applied_taxes": null,
      "converting_from_quote": false,
      "item_applied_taxes": null,
      "shipping_assignments": [
        {
          "items": [
            {
              "additional_data": "",
              "base_cost": 45,
              "base_original_price": 45,
              "base_price": 45,
              "base_price_incl_tax": null,
              "base_row_total": 45,
              "base_row_total_incl_tax": 48.71,
              "base_tax_amount": 3.71,
              "created_at": null,
              "item_id": null,
              "name": "Push It Messenger Bag",
              "order_id": null,
              "original_price": 45,
              "price": 45,
              "price_incl_tax": 48.71,
              "product_id": 14,
              "product_type": "simple",
              "qty_ordered": 1,
              "row_total": 45.51,
              "row_total_incl_tax": 48.71,
              "sku": "24-MB06",
              "store_id": 1,
              "tax_amount": 3.71,
              "updated_at": null
            } ],
          "shipping": {
            "address": {
              "address_type": "shipping",
              "city": "Lynnwood,",
              "company": "",
              "country_id": "US",
              "customer_address_id": null,
              "email": "roni_cost@example.com",
              "entity_id": null,
              "firstname": "Veronica",
              "lastname": "Costello",
              "parent_id": null,
              "postcode": "98087-6266",
              "region": "WA",
              "region_code": null,
              "street": [
                "13724 39th Pl W",
                ""
              ],
              "telephone": "0000000000"
            },
            "method": "flatrate_flatrate",
            "total": {
              "base_shipping_amount": 5,
              "base_shipping_discount_amount": 0,
              "base_shipping_discount_tax_compensation_amnt": 0,
              "base_shipping_incl_tax": 5,
              "base_shipping_tax_amount": 0,
              "shipping_amount": 5,
              "shipping_discount_amount": 0,
              "shipping_discount_tax_compensation_amount": 0,
              "shipping_incl_tax": 5,
              "shipping_tax_amount": 0
            } }
        } ]
    },
    "global_currency_code": "USD",
    "grand_total": 49.90,
    "increment_id": null,
    "items": [
      {
        "additional_data": "232",
        "base_cost": 45,
        "base_original_price": 45,
        "base_price": 45,
        "base_price_incl_tax": null,
        "base_row_total": 45,
        "base_row_total_incl_tax": 48.71,
        "base_tax_amount": 3.71,
        "created_at": null,
        "item_id": null,
        "name": "Push It Messenger Bag",
        "order_id": null,
        "original_price": 45,
        "price": 45,
        "price_incl_tax": null,
        "product_id": 14,
        "product_type": "simple",
        "qty_ordered": 1,
        "row_total": 45,
        "row_total_incl_tax": 48.71,
        "sku": "24-WB04",
        "store_id": 1,
        "tax_amount": 3.71,
        "updated_at": null
      } ],
    "order_currency_code": "USD",
    "payment": {
      "account_status": null,
      "additional_information": null,
      "amount_ordered": 53.71,
      "amount_paid": 0,
      "base_amount_ordered": 53.71,
      "base_amount_paid": 0,
      "base_shipping_amount": 5.00,
      "cc_last4": null,
      "entity_id": null,
      "method": "checkmo",
      "parent_id": null,
      "shipping_amount": 5.00
    },
    "protect_code": null,
    "shipping_amount": 5.00,
    "shipping_description": "Ground - 4-7 Business Days",
    "shipping_discount_amount": null,
    "shipping_discount_tax_compensation_amount": 0,
    "shipping_incl_tax": 5.00,
    "shipping_tax_amount": null,
    "state": "new",
    "status": "pending",
    "status_histories": null,
    "store_currency_code": "USD",
    "store_id": 1,
    "store_name": null,
    "store_to_base_rate": 0,
    "store_to_order_rate": 0,
    "subtotal": 45.00,
    "subtotal_incl_tax": 48.71,
    "tax_amount": 3.71,
    "total_due": null,
    "total_item_count": 1,
    "total_paid": null,
    "total_qty_ordered": 1,
    "updated_at": null
  }
}
```

### Steps to reproduce (*)

1. Place order with product which has the enabled "Manage Stock" option. In this example, I've used - `24-WB04`

### Expected result (*)

1. Order is created via REST API call. Ordered item is reserved and exists in the `inventory_reservation` table

### Actual result (*)

1. Order is created but ordered item is not reserved and does not exist in the `inventory_reservation` table

### PR DESCRIPTION
**Pull request by atwixfirster**
### Description (*)

PR fixes https://github.com/magento/magento2/issues/26575.

### Fixed Issues (if relevant)

1. https://github.com/magento/magento2/issues/26575: REST API rest/all/V1/orders/create does not trigger "inventory_reservations_placement" plugin

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by dmytro-ch** [comment]
Please note, this PR provides backward-incompatible change, due to the different parameter names.

**Comment by atwixfirster** [comment]
> Please note, this PR provides backward-incompatible change, due to the different parameter names.

Thanks for notice, @dmytro-ch !

Then I've to refactor my fix.

Thanks again!

**Comment by atwixfirster** [comment]
@dmytro-ch , could you please review a new solution?

THank you!

File: app/code/Magento/Sales/Api/OrderPlacementInterface.php [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+namespace Magento\Sales\Api;
+
+use Magento\Sales\Api\Data\OrderInterface;
+
+/**
+ * Introduce an interface to place order via REST API
+ * @api
+ */
+interface OrderPlacementInterface
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Api/OrderPlacementInterface.php:16]:
```suggestion
interface PlaceOrderInterface
```

File: app/code/Magento/Sales/Model/Service/OrderPlacementService.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\Sales\Model\Service;
+
+use Magento\Sales\Api\OrderPlacementInterface;
+use Exception;
+use Magento\Sales\Api\Data\OrderInterface;
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Model/Service/OrderPlacementService.php:12]: Please sort imports.

File: app/code/Magento/Sales/Model/Service/OrderPlacementService.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * @inheritDoc
+ */
+class OrderPlacementService implements OrderPlacementInterface
```
Comment (dmytro-ch) [inline app/code/Magento/Sales/Model/Service/OrderPlacementService.php:17]:
```suggestion
class PlaceOrderService implements PlaceOrderInterface
```

**Review by dmytro-ch — CHANGES_REQUESTED** [review]
@atwixfirster, thank you for the updates!
Could you please change the class naming a little bit?

Please note, due to [Magento Definition of Done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html#automated-tests) the changes must be covered by auto tests. Could you please cover your case by API functional test?

Thank you!

**Comment by VladimirZaets** [comment]
@atwixfirster , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for collaboration

**Comment by atwixfirster** [comment]
> @atwixfirster , I am closing this PR now due to inactivity.
> Please reopen and update if you wish to continue.
> Thank you for collaboration

Hi, @VladimirZaets !

Unfortunately,  I do not have access to reopen a PR. Can you please reopen it?

Thanks!

File: app/code/Magento/Sales/Model/Service/OrderPlacementService.php (hunk 3 above)
Comment (atwixfirster) [reply app/code/Magento/Sales/Model/Service/OrderPlacementService.php:17]: **done**

File: app/code/Magento/Sales/Model/Service/OrderPlacementService.php (hunk 2 above)
Comment (atwixfirster) [reply app/code/Magento/Sales/Model/Service/OrderPlacementService.php:12]: **done**

File: app/code/Magento/Sales/Api/OrderPlacementInterface.php (hunk 1 above)
Comment (atwixfirster) [reply app/code/Magento/Sales/Api/OrderPlacementInterface.php:16]: **done**

**Pull request closed without merging**